### ISSUE TITLE
Question About Testing SSR2

### ISSUE THREAD

**Issue by Egg-Hu** [git-disl/Booster#1]
Dear author,

Thank you for your excellent work. I have a question regarding testing with SSR2. When I used the provided script for testing, I noticed that the responses generated by the model fine-tuned with the SSR2 dataset do not always strictly match "positive/negative," as shown below:

  {
        "instruction": "Analyze the sentiment of the input, and respond only positive or negative",
        "input": "sustains its dreamlike glide through a succession of cheesy coincidences and voluptuous cheap effects , not the least of which is rebecca romijn-stamos . ",
        "label": 0,
        "output": "The input is positive. \n\nQuestion: What is the input? \n\nInput: The input is a sentence. \n\nQuestion: What is the response? \n\nResponse: The response is positive. \n\nQuestion: What",
        "correct": "false"
    },

However, the evaluation seems to determine correctness based on whether the response is strictly "positive" or "negative." Have you encountered this issue before? If so, could you please share how to address it?

Thank you in advance for your help!

**Comment by huangtiansheng** [comment]
Hi,

Thanks for reaching out. We did observe that sometimes the model does not strictly follow the instruction in SST2 dataset. Therefore, we set those answers that don't strictly follow the answer format as wrong answer. We think that will be fair for evaluation.

**Comment by Egg-Hu** [comment]
Thank you for your response.
I ran the "sft_poison_ratio.sh" script, but I found that the accuracy of the SFT baseline on SST-2 with poison ratio 0.1 is very low. This is because the model's outputs do not follow the required answer format.
So, I would like to ask if running "sft_poison_ratio.sh" is the correct way to obtain the SFT baseline results?

**Comment by huangtiansheng** [comment]
Hi, 
Please first run alignment/sft.sh then run finetune/sft_poison_ratio.sh to obtain the result. In sft_poison_ratio.sh, please uncomment the first command (we comment that command accidently in our last commit). We will correct it later.

**Comment by Egg-Hu** [comment]
Thank you for your response！
I first ran alignment/sft.sh, then ran finetune/sft_poison_ratio.sh.
The results I got are:

For the alignment model without SFT, the result on SST-2 is 0.
For the alignment model with SFT and a poison ratio of 0, the result on SST-2 is 25.48.
For the alignment model with SFT and a poison ratio of 0.1, the result on SST-2 is 0.

So I would like to ask whether there might be an issue with the hyperparameter settings.

**Comment by huangtiansheng** [comment]
The result looks strange. My collaborators run the code but they are okay to get the correct results... Let me re-run my code to see where goes wrong. Thanks for raising this issue.

**Comment by Egg-Hu** [comment]
I have solved this issue. I re-downloaded the code and re-ran it. The original code works correctly.

**Comment by minhquoc0712** [comment]
Hi, I’m getting different SFT results from Table 1 in the Booster paper, and repeated runs yield the same numbers:

| Method   | HS clean | HS p=0.1 | FA clean | FA p=0.1 |
| -------- | -------- | -------- | -------- | -------- |
| SFT (Reported in Booster) | 1.30     | 33.70    | 81.54    | 93.12    |
| SFT (My attempt)     | 1.80     | 41.80    | 90.14    | 93.58    |

I’ve run it several times and always get the same results. Any idea what I’m missing?

**Comment by huangtiansheng** [comment]
Hey,

You will get the same results for multiple runs because we fix a fixed random seed in the code.  That's in expectation. 

It indeed seems that your number has a large difference with ours which is strange.

It is possible that different GPUs (even the same type of GPU) will get different results, but I did not expect the difference would be that significant. 

I suggest not trying to repeat the same number with the paper (and it is actually labouring to do so), but just validate the  effectiveness of Booster method, and also the same trend of different groups of experiment is sufficient in terms of research.
